### PR TITLE
1-D index下的paddle.gather映射由narrow+squeeze+stack改成torch.index_select

### PR DESCRIPTION
**Pull request by omoYang**
paddle.gather 的 1-D index（非标量、非空）对应的torch API由逐元素narrow/squeeze/stack 循环改为 torch.index_select，语义等价

torch=2.12.1+cu130 cuda=13.0
| shape+index+axis+n_idx | fwd | bwd |
|---|---|---|
| shape=(1, 3, 512, 64) idx=[0, 0, 0] axis=1 n_idx=3 | OK | MISMATCH max_diff=4.76837e-07 n_diff=10460 |
| shape=(1, 3, 512, 64) idx=[0, 0] axis=1 n_idx=2 | OK | OK |
| shape=(1, 3, 512, 64) idx=[0] axis=1 n_idx=1 | OK | OK |

当index中同一位置重复出现3times+时，受torch.stack（逆向遍历index）和torch.index_select（正向遍历index）反向中遍历顺序不同的影响，浮点数不满足结合律而导致结果之间有误差，其余时候逐元素相等，因此可以替换

### CODE REVIEW

**Review by cangtianhuang — APPROVED** [review]
LGTM
